go-controller/pkg/ovn/pods.go

```diff
@@ -381,6 +405,54 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 		}
 	}
 
+	// add src-ip routes to GR if external gw annotation is set
+	routingExternalGWs, err := oc.getRoutingExternalGWs(pod.Namespace)
+	if err != nil {
+		return err
+	}
+	if routingExternalGWs != nil {
+		podIP := podIfAddrs[0].IP.String()
+		gr := "GR_" + pod.Spec.NodeName
+		for _, v := range routingExternalGWs {
+			gw := v.String()
+			_, stderr, err := util.RunOVNNbctl("--may-exist", "--policy=src-ip", "--ecmp",
+				"lr-route-add", gr, podIP+"/32", gw)
+			if err != nil {
+				return fmt.Errorf("unable to add external gw src-ip route to GR router, stderr:%q, err:%v", stderr, err)
+			}
+			nsInfo, err := oc.waitForNamespaceLocked(pod.Namespace)
+			if err != nil {
+				return err
+			}
+			nsInfo.podExternalRoutes[podIP] = append(nsInfo.podExternalRoutes[podIP], externalRoute{gr: gr, gw: gw})
+			nsInfo.Unlock()
+		}
+	}
+	// add src-ip routes to GR if routing namespace annotation is set
+	if val, ok := pod.Annotations["k8s.ovn.org/routing-namespace"]; ok {
+		// validate namespace exists
+		if pod.Spec.HostNetwork {
```

Comment:
need to check for the routing-network annotation and find the multus annotation IPs